### PR TITLE
Adding truss for code llama 7b

### PR DESCRIPTION
**Pull request by htrivedi99**

### CODE REVIEW

File: code-llama-7b-instruct/model/model.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    def __init__(self, data_dir: str, config: Dict, **kwargs):
+        self._data_dir = data_dir
+        self._config = config
+        self.model = None
+        self.tokenizer = None
+
+    def preprocess(self, request: dict):
+        generate_args = {
+            "max_new_tokens": 128,
```
Comment (squidarth) [inline code-llama-7b-instruct/model/model.py:22]: nit: could we have this be:
```
"max_new_tokens": request.get("max_tokens", 128)
```

instead? (and ditto for the other parameters)

File: code-llama-7b-instruct/model/model.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                formatted_prompt = f"<s><<SYS>>\n{DEFAULT_SYSTEM_PROMPT}\n<</SYS>>\n\n{prompt}"
+                input_ids = self.tokenizer(
+                    formatted_prompt, return_tensors='pt').input_ids.cuda()
+                output = self.model.generate(
+                    inputs=input_ids,
+                    **request["generate_args"]
+                )
+
+                return self.tokenizer.decode(output[0])
+            except Exception as exc:
```
Comment (squidarth) [inline code-llama-7b-instruct/model/model.py:68]: I don't think we need this try/except -- exceptions will be handled by Truss and turned into the correct errors

**Review by squidarth — COMMENTED** [review]
This looks great! Left a couple nits.

One q: just to double check, code llama doesn't require some kind of special license, does it?

File: code-llama-7b-instruct/model/model.py (hunk 1 above)
Comment (htrivedi99) [reply code-llama-7b-instruct/model/model.py]: Good suggestion. I updated the code.

File: code-llama-7b-instruct/model/model.py (hunk 2 above)
Comment (htrivedi99) [reply code-llama-7b-instruct/model/model.py]: Ok, try except block removed

**Comment by htrivedi99** [comment]
> This looks great! Left a couple nits.
> 
> One q: just to double check, code llama doesn't require some kind of special license, does it?



> This looks great! Left a couple nits.
> 
> One q: just to double check, code llama doesn't require some kind of special license, does it?

When you say special license do you mean permission to access the model?

**Comment by squidarth** [comment]
> When you say special license do you mean permission to access the model?

Yeah -- exactly

File: code-llama-7b-instruct/model/model.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        )
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            self.MODEL_NAME
+        )
+
+    def predict(self, request: Dict) -> Dict:
+        with torch.no_grad():
+            prompt = request.pop("prompt")
+            # Code-llama needs the prompt to be formatted in this manner
+            formatted_prompt = f"<s><<SYS>>\n{DEFAULT_SYSTEM_PROMPT}\n<</SYS>>\n\n{prompt}"
```
Comment (varunshenoy) [inline code-llama-7b-instruct/model/model.py:50]: Can we also template out the default system prompt? Would be good to give users a way to pass in their own system prompt.

**Review by varunshenoy — COMMENTED** [review]
Awesome, just a small nit.

**Comment by htrivedi99** [comment]
> > When you say special license do you mean permission to access the model?
> 
> Yeah -- exactly

No, code llama does not require any special license

**Comment by squidarth** [comment]
Great! thanks again for the contributioN!

**Pull request merged**